### PR TITLE
Feature/condo/sberdoma 20/ticket card

### PR DESCRIPTION
**Pull request by Dimitreee**
Screenshot 2021-02-24 at 14 54 04

### CODE REVIEW

File: apps/condo/utils/clientSchema/Ticket/Status.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+function convertGQLItemToUIState (item) {
+    if (item.dv !== 1) throw new Error('unsupported item.dv')
+    return { ...item, dv: undefined }
+}
+
+function convertUIStateToGQLItem (item) {
+    return item
+}
+
+function convertGQLItemToFormSelectState (item) {
```
Comment (Dimitreee) [inline apps/condo/utils/clientSchema/Ticket/Status.ts:13]: @pahaz is it ok to place function's like this here in `clientScheema` utils, or should we delegate this work to single `convertGQLItemToUIState()` ?

File: apps/condo/utils/clientSchema/search.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import {
```
Comment (Dimitreee) [inline apps/condo/utils/clientSchema/search.ts:1]: @pahaz I want to introduce  `clientSchema/serach` module which will include all search functions needed

File: apps/condo/lang/ru.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
   "pages.condo.property.type.building": "Многоквартирный дом",
   "pages.condo.property.type.village": "Коттеджный поселок",
   "pages.condo.ticket.index.PageTitle": "Задачи",
+  "pages.condo.ticket.id.PageTitle": "Задача",
```
Comment (pahaz) [inline apps/condo/lang/ru.json:22]:
```suggestion
  "pages.condo.ticket.id.PageTitle": "Заявка",
```

File: apps/condo/lang/ru.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
   "pages.condo.property.type.building": "Многоквартирный дом",
   "pages.condo.property.type.village": "Коттеджный поселок",
   "pages.condo.ticket.index.PageTitle": "Задачи",
```
Comment (pahaz) [inline apps/condo/lang/ru.json:21]:
```suggestion
  "pages.condo.ticket.index.PageTitle": "Заявки",
```

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 5]
```diff
@@ -1,88 +1,192 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
-import { Descriptions, Form } from 'antd'
-import React from 'react'
+import { Button, Select, Row, Col, Typography, Statistic, Divider } from 'antd'
+import get from 'lodash/get'
+import React, { useMemo, useCallback } from 'react'
+import { format } from 'date-fns'
+
+// TODO: move to packages later
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:6]:
```suggestion
// TODO(Dimitreee): move to packages later
```

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 6]
```diff
@@ -1,88 +1,192 @@
...
-
 import { useIntl } from '@core/next/intl'
-
 import { PageContent, PageHeader, PageWrapper } from '../../../containers/BaseLayout'
-
 import LoadingOrErrorPage from '../../../containers/LoadingOrErrorPage'
-import { useObject, useUpdate } from '../../../schema/Ticket.uistate'
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// TODO: Add types to utils clientScheema ddanew
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:17]:
```suggestion
// TODO(Dimitreee): Add types to utils clientScheema ddanew
```

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 7]
```diff
@@ -1,88 +1,192 @@
...
+        en: EN,
+    }
+
+    const formattedCreatedDate = format(
+        new Date(ticket.createdAt),
+        'dd MMMM (HH:mm)',
+        { locale: locales[intl.locale] }
+    )
+
+    // TODO: rewrite to template string
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:58]:
```suggestion
    // TODO(Dimitreee): rewrite to a template string
```

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 8]
```diff
@@ -1,88 +1,192 @@
...
 
     const router = useRouter()
     const { query: { id } } = router
-    const { refetch, loading, obj, error } = useObject({ where: { id } })
-    const update = useUpdate({}, () => refetch())
+    const { refetch, loading, obj: ticket, error } = Ticket.useObject({ where: { id } })
+    const update = Ticket.useUpdate({}, () => refetch())
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    // TODO: add types to runMutation ~ ddanew
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:106]:
```suggestion
    // TODO(Dimitreee): add types to runMutation ~ ddanew
```

File: apps/condo/utils/clientSchema/Ticket/Status.ts (hunk 1 above)
Comment (pahaz) [reply apps/condo/utils/clientSchema/Ticket/Status.ts:13]: I think it's ok. I'm a little confused by the naming. `item` is the result of `convertGQLItemToUIState` it's `UIState`. maybe it's better to name it like `contertToFormSelectState` or `convertUIStateToFormSelectState` or etc...

Comment (pahaz) [reply apps/condo/utils/clientSchema/Ticket/Status.ts:13]: There is one more important point. What we need to do if we needed to define a `convertGQLItemToFormSelectState` function for different models. For example for `TicketStatus` and `TicketSource`. Maybe it's better to export it as: 
```
const TicketStatus = genReactHooks(TicketStatus, { convertGQLItemToUIState, convertUIStateToGQLItem })
TicketStatus.convertGQLItemToFormSelectState = ...
module.exports = {
    TicketStatus,
}
```

File: apps/condo/utils/clientSchema/search.ts (hunk 2 above)
Comment (pahaz) [reply apps/condo/utils/clientSchema/search.ts:1]: I have some thoughts on this.

File: apps/condo/utils/clientSchema/search.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    GET_ALL_CLASSIFIERS_QUERY,
+    GET_ALL_ORGANIZATION_EMPLOYEE_QUERY,
+    GET_ALL_PROPERTIES_QUERY,
```
Comment (pahaz) [inline apps/condo/utils/clientSchema/search.ts:4]: It confuses me that we have auto-generated queries with a similar name. 
`Ticket.GET_ALL_OBJS_QUERY` with all fields.

File: apps/condo/utils/clientSchema/search.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+} from '../../schema/Ticket.gql'
+
+async function _search (client, query, variables) {
+    return await client.query({
+        query: query,
+        variables: variables,
+    })
+}
+
+export async function searchProperty (client, value) {
```
Comment (pahaz) [inline apps/condo/utils/clientSchema/search.ts:15]: Maybe it's better to put in near Model

File: apps/condo/utils/clientSchema/Ticket/TicketStatus.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+const {
+    useObject,
+    useObjects,
+    useCreate,
+    useUpdate,
+    useDelete,
+} = genReactHooks(TicketStatus, { convertGQLItemToUIState, convertUIStateToGQLItem })
+
+export {
+    useObject,
```
Comment (pahaz) [inline apps/condo/utils/clientSchema/Ticket/TicketStatus.ts:32]: May be
```
const TicketStatus = genReactHooks(TicketStatus, { convertGQLItemToUIState, convertUIStateToGQLItem })
exxport TicketStatus
```

Comment (pahaz) [reply apps/condo/utils/clientSchema/Ticket/TicketStatus.ts:32]: similar to `TicketStatus.js`

**Pull request merged**